system/jlib/jdebug.cpp

```diff
@@ -1967,10 +1967,17 @@ class CExtendedStats  // Disk network and cpu stats
         memset(&oldnet, 0, sizeof(oldnet));
         memset(&newnet, 0, sizeof(newnet));
         ndisks = 0;
+        kbufmax = 0;
+#ifdef __linux__
         if (printklog)
-            kbufmax = 0x1000;
-        else
-            kbufmax = 0;
+        {
+            // kernel ring buffer size - supported since 2.6.6
+            // ignore kern msgs for anything older
+            int klogmax = klogctl(10, NULL, 0);
```

Comment:
Don't you need to do this inside getKLog ? i.e. each time it is called - as size of kernel ring buffer may have increased?
Or is it the max size of the kernel ring buffer, e.g. even if empty ?
Even so, I guess could change between calls?